app/modules/conversations/conversation/conversation_controller.py

```
@@ -188,6 +188,22 @@ async def get_last_human_message(self, conversation_id: str):
         except ConversationServiceError as e:
             raise HTTPException(status_code=500, detail=str(e))
 
+    async def update_agent(
+        self, conversation_id: str, agent_id: str
+    ) -> ConversationInfoResponse:
+        try:
+            return await self.service.update_conversation_agent(
+                conversation_id, agent_id, self.user_id
+            )
+        except ConversationNotFoundError as e:
+            raise HTTPException(status_code=404, detail=str(e))
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)